### PR TITLE
drivers:dac:ltc2688: Refactor LTC2688 to suit ltc2686/8.

### PR DESCRIPTION
**Pull request by pmallick-adi**
* Replaced the use of "LTC2688" with "LTC268X" in functions, variables and macros.
* Added device ids for LTC2686/8.
* Updated command word to support channel configurations for ltc2686.
* Renamed the file and folder names.

Signed-off-by: Pratyush Mallick <Pratyush.Mallick@analog.com>

### CODE REVIEW

**Review by amiclaus — COMMENTED** [review]
in order to ease the review process, these modifications should be splitted into multiple commits.
- one for the renamings (files + functions + variables) - LTC268X
- one for adding support for multiple devices - LTC2686/8

**Comment by pmallick-adi** [comment]
> in order to ease the review process, these modifications should be splitted into multiple commits.
> 
> * one for the renamings (files + functions + variables) - LTC268X
> * one for adding support for multiple devices - LTC2686/8

Done.

File: drivers/dac/ltc268x/ltc268x.h [hunk 1]
```diff
@@ -47,15 +47,19 @@
 /******************************************************************************/
 /******************* Macros and Constants Definitions *************************/
 /******************************************************************************/
-#define LTC268X_DAC_CHANNELS	16
+#define LTC2686_DAC_CHANNELS          8
+#define LTC2688_DAC_CHANNELS          16
+#define LTC268X_DAC_CHANNELS(id)	  (id ? LTC2688_DAC_CHANNELS : LTC2686_DAC_CHANNELS)
```
Comment (amiclaus) [inline drivers/dac/ltc268x/ltc268x.h:52]: I'm thinking about adding in the future more devices compatible with this driver.

In that case this macro won't be the best approach.

I'd rather create an array storing these values based on the index (which will be the device id defined with `enum ltc268x_device_id`

**Review by amiclaus — COMMENTED** [review]
one inline comment.

Also please add commit description to 60e299e .

Otherwise, lgtm, ready to be merged.

File: drivers/dac/ltc268x/ltc268x.h (hunk 1 above)
Comment (pmallick-adi) [reply drivers/dac/ltc268x/ltc268x.h:52]: I agree. Will implement this.

File: drivers/dac/ltc268x/ltc268x.c [hunk 2]
```diff
@@ -180,20 +180,21 @@ int32_t ltc2688_set_dither_toggle(struct ltc2688_dev *dev, uint16_t setting)
...
  */
-int32_t ltc2688_set_dither_mode(struct ltc2688_dev *dev, uint8_t channel,
+int32_t ltc268x_set_dither_mode(struct ltc268x_dev *dev, uint8_t channel,
 				bool en)
 {
 	uint16_t val = 0;
 	int32_t ret;
 
-	if (channel >= LTC2688_DAC_CHANNELS)
+	if (channel >= ltc268x_channels(dev->dev_id))
```
Comment (rbolboac) [inline drivers/dac/ltc268x/ltc268x.c:189]: seeing how ltc268x_channels is now an array, should be accessed using []: ltc268x_channels[dev->dev_id]
apply to the whole file

Comment (pmallick-adi) [reply drivers/dac/ltc268x/ltc268x.c]: this has been done

**Review by rbolboac — APPROVED** [review]
lgtm

**Comment by pmallick-adi** [comment]
Can we get this merged?

**Review by amiclaus — APPROVED** [review]

**Pull request merged**